src/common/bcknd/device/opencl/rhs_maker.c

```diff
@@ -167,3 +197,38 @@ void fluid_makebdf_opencl(void *ulag1, void *ulag2, void *vlag1,
                                   0, NULL, NULL));
   
 }
+
+void scalar_rhs_maker_bdf_opencl(void *s_lag, void *s_laglag, void *fs,
+                                 void *s, void *B, real *rho, real *dt,
+                                 real *bd2, real *bd3, real *bd4,
+                                 int *nbd, int *n) {
+  cl_int err;
+  
+  if (rhs_maker_program == NULL)
+    opencl_kernel_jit(rhs_maker_kernel, (cl_program *) &rhs_maker_program);
+
+  cl_kernel kernel = clCreateKernel(rhs_maker_program, "scalar_makebdf_kernel", &err);
+  CL_CHECK(err);
+
+  CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), (void *) &s_lag));
+  CL_CHECK(clSetKernelArg(kernel, 1, sizeof(cl_mem), (void *) &s_laglag));
+  CL_CHECK(clSetKernelArg(kernel, 6, sizeof(cl_mem), (void *) &fs));
+  CL_CHECK(clSetKernelArg(kernel, 9, sizeof(cl_mem), (void *) &s));
+  CL_CHECK(clSetKernelArg(kernel, 12, sizeof(cl_mem), (void *) &B));
+  CL_CHECK(clSetKernelArg(kernel, 13, sizeof(real), rho));
+  CL_CHECK(clSetKernelArg(kernel, 14, sizeof(real), dt));
+  CL_CHECK(clSetKernelArg(kernel, 15, sizeof(real), bd2));
+  CL_CHECK(clSetKernelArg(kernel, 16, sizeof(real), bd3));
+  CL_CHECK(clSetKernelArg(kernel, 17, sizeof(real), bd4));
+  CL_CHECK(clSetKernelArg(kernel, 18, sizeof(int), nbd));
+  CL_CHECK(clSetKernelArg(kernel, 19, sizeof(int), n));
```

Comment:
```suggestion
  CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), (void *) &s_lag));
  CL_CHECK(clSetKernelArg(kernel, 1, sizeof(cl_mem), (void *) &s_laglag));
  CL_CHECK(clSetKernelArg(kernel, 2, sizeof(cl_mem), (void *) &fs));
  CL_CHECK(clSetKernelArg(kernel, 3, sizeof(cl_mem), (void *) &s));
  CL_CHECK(clSetKernelArg(kernel, 4, sizeof(cl_mem), (void *) &B));
  CL_CHECK(clSetKernelArg(kernel, 5, sizeof(real), rho));
  CL_CHECK(clSetKernelArg(kernel, 6, sizeof(real), dt));
  CL_CHECK(clSetKernelArg(kernel, 7, sizeof(real), bd2));
  CL_CHECK(clSetKernelArg(kernel, 8, sizeof(real), bd3));
  CL_CHECK(clSetKernelArg(kernel, 9, sizeof(real), bd4));
  CL_CHECK(clSetKernelArg(kernel, 10, sizeof(int), nbd));
  CL_CHECK(clSetKernelArg(kernel, 11, sizeof(int), n));
```

Must match the number of arguments in the kernel, otherwise it will cause a runtime error during JIT+run